piksi_multi_cpp/src/device/device_usb.cc

```diff
@@ -7,7 +7,17 @@ const int kInterfaceNumber = 1;
 
 namespace piksi_multi_cpp {
 
-DeviceUSB::DeviceUSB(const Identifier& id) : Device(id), port_(nullptr) {}
+DeviceUSB::DeviceUSB(const Identifier& id) : DeviceSerial(id) {}
+
+bool DeviceUSB::parseId() {
+  // not much to check with USB ids.
```

Comment:
I guess opening a USB device by port, e.g. `/dev/ttyACM0 ` will default automatically to serial device. Nice.

Reply:
hmm if you use serial:///dev/ttyACM0 yes. if you use usb:///dev/ttyACM0 it would look for a device with serial number "/dev/ttyACM0" 

If piksi serial numbers have some pattern we could also add a check there and throw an error if its not a valid S/N

Reply:
LGTM